cfgov/v1/atomic_elements/atoms.py

```diff
@@ -154,3 +155,33 @@ def clean(self, data):
             raise ValidationError("ImageBasicUrlAlt validation errors", params=error_dict)
         else:
             return data
+
+class AnchorLink(blocks.StructBlock):
+    link_id = blocks.CharBlock(required=False)
+    # full_url = blocks.CharBlock(required=False, default=slugurl)
+
+    def clean(self, data):
+        error_dict = {}
+
+        def format_id(string):
+            words = string.split('_');
+            suffix = ''
+            if string:
+                suffix = '_'
+            if 'anchor' in words:
+                return slugify(string)
+            else:
+                return get_unique_id('anchor_' + slugify(string) + suffix)
```

Comment:
If I understand this right, you could make a few adjustments to this, like so:

```python
def format_id(string):
    if 'anchor' in string:
        return slugify(string)
    else:
        suffix = '_' if string else ''
        return get_unique_id('anchor_' + slugify(string) + suffix)
```
What's the point of `suffix`?